src/Data/Fin/Substitution/Lemmas.agda

```diff
@@ -543,3 +543,69 @@ record TermLemmas (T : ℕ → Set) : Set₁ where
     T.weaken t₁ T./ (T.sub t₂) ≡⟨ weaken-∷ t₁ ⟩
     t₁ T./ T.id                ≡⟨ id-vanishes t₁ ⟩
     t₁                         ∎
+
+  -- Lemmas relating renamings to substitutions.
+
+  map-var≡ :
+    ∀ {m n} {ρ₁ : Sub Fin m n} {ρ₂ : Sub T m n} {f : Fin m → Fin n} →
+    (∀ x → lookup ρ₁ x ≡ f x) →
+    (∀ x → lookup ρ₂ x ≡ T.var (f x)) →
+    map T.var ρ₁ ≡ ρ₂
+  map-var≡ {ρ₁ = ρ₁} {ρ₂ = ρ₂} {f = f} hyp₁ hyp₂ = extensionality λ x →
+    lookup (map T.var ρ₁) x  ≡⟨ VecProp.lookup-map x _ ρ₁ ⟩
+    T.var (lookup ρ₁ x)      ≡⟨ cong T.var $ hyp₁ x ⟩
+    T.var (f x)              ≡⟨ sym $ hyp₂ x ⟩
+    lookup ρ₂ x              ∎
+
+  wk≡wk : ∀ {n} → map T.var VarSubst.wk ≡ T.wk {n = n}
```

Comment:
Also no idea, if whoever's in charge of this refactoring is up for this, but if you wanted and think it would help readability, then it would be a good opportunity to add some `variable`s to this file to remove some of the repetition in the types?

Reply:
That's a great suggestion, but for a subsequent revision.